lib/gooddata/models/segment.rb

```diff
@@ -43,10 +43,13 @@ def [](id, opts = {})
         client = domain.client
         fail ArgumentError, 'No client specified' if client.nil?
 
+        data_product = opts[:data_product]
+
         if id == :all
           GoodData::Segment.all(opts)
         else
-          result = client.get(domain.segments_uri + "/segments/#{CGI.escape(id)}")
+          uri = data_product ? GoodData::DataProduct::ONE_DATA_PRODUCT_PATH % { domain_name: domain.name, id: data_product.data_product_id } : domain.segments_uri
```

Comment:
Refers to 'domain' more than self (maybe move it to another class?) (FeatureEnvy)